### PR TITLE
Add hash to cluster images

### PR DESCRIPTION
**Pull request by davidvossel**
kubevirtci rebuilds their images at times to experiment with different changes. In order to guarantee the image we consume is in fact immutable, we must consume the image with the sha hash.

### CODE REVIEW

File: hack/build/config.sh [hunk 1]
```diff
@@ -35,8 +35,8 @@ function allPkgs {
     echo "$ret"
 }
 
-KUBERNETES_IMAGE="k8s-1.10.4"
-OPENSHIFT_IMAGE="os-3.10.0"
+KUBERNETES_IMAGE="k8s-1.10.4@sha256:09ac918cc16f13a5d0af51d4c98e3e25cbf4f97b7b32fe18ec61b32f04ca1009"
+OPENSHIFT_IMAGE="os-3.10.0@sha256:14ffc4a28e24a2510c9b455b56f35f6193a00b71c9150705f6afec41b003fc76"
```
Comment (jeffvance) [inline hack/build/config.sh]: where is this used and why?

Comment (copejon) [inline hack/build/config.sh]: I can't check `make cluster-up` myself being on a mac - does this cause any trouble w/ the `case` in https://github.com/kubevirt/containerized-data-importer/blob/584b2eca79bafd75f6ae3198599d1ed19e61b566/cluster/up.sh#L12 ?

**Comment by mhenriks** [comment]
LGTM

File: hack/build/config.sh (hunk 1 above)
Comment (jeffvance) [reply hack/build/config.sh]: @davidvossel comment/thoughts on Jon's question?

Comment (davidvossel) [reply hack/build/config.sh]: it shouldn't. I ran 'make cluster-up' with this change and didn't experience any issues.

Comment (davidvossel) [reply hack/build/config.sh]: i added k8s-10.4 as the default provider unless the user overrides that env var.

**Pull request merged**